lib/sync.coffee

```diff
@@ -166,7 +166,14 @@ module.exports =
 		resin.models.device.isOnline(params.uuid).tap (isOnline) ->
 			throw new Error('Device is not online') if not isOnline
 		.then ->
-			return resin.models.device.get(params.uuid).get('uuid').then(performSync)
+			return resin.models.device.hasDeviceUrl(params.uuid)
+		.then (hasDeviceUrl) ->
+			if hasDeviceUrl
+				return resin.models.device.get(params.uuid).get('uuid').then(performSync)
+			else
+				return resin.models.device.get(params.uuid).get('uuid').then (fullUUID) ->
+					return resin.models.device.enableDeviceUrl(fullUUID).then ->
+						return resin.models.device.get(params.uuid).get('uuid').delay(2000).then(performSync)
```

Comment:
You already have access to the full UUID with the `fullUUID` variable above, so no need to do `resin.models.device.get()` again here.